heron/packing/tests/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPackingTest.java

```
@@ -0,0 +1,570 @@
+    // Setup the spout parallelism
+    Map<String, Integer> spouts = new HashMap<>();
+    spouts.put(SPOUT_NAME, spoutParallelism);
+
+    // Setup the bolt parallelism
+    Map<String, Integer> bolts = new HashMap<>();
+    bolts.put(BOLT_NAME, boltParallelism);
+
+    TopologyAPI.Topology topology =
+        TopologyTests.createTopology("testTopology", topologyConfig, spouts, bolts);
+
+    return topology;
+  }
+
+  protected PackingPlan getResourceCompliantRRPackingPlan(TopologyAPI.Topology topology) {
+    Config config = Config.newBuilder()
+        .put(Keys.topologyId(), topology.getId())
+        .put(Keys.topologyName(), topology.getName())
+        .putAll(ClusterDefaults.getDefaults())
+        .build();
+
+    this.instanceRamDefault = Context.instanceRam(config);
+    this.instanceCpuDefault = Context.instanceCpu(config).doubleValue();
+    this.instanceDiskDefault = Context.instanceDisk(config);
+
+    Config runtime = Config.newBuilder()
+        .put(Keys.topologyDefinition(), topology)
+        .build();
+
+    ResourceCompliantRRPacking packing = new ResourceCompliantRRPacking();
+    packing.initialize(config, runtime);
+    PackingPlan output = packing.pack();
+
+    return output;
+  }
+
+  @Test
+  public void testCheckFailure() throws Exception {
+    int numContainers = 2;
+    int spoutParallelism = 4;
+    int boltParallelism = 3;
+
+    // Set up the topology and its config
+    com.twitter.heron.api.Config topologyConfig = new com.twitter.heron.api.Config();
+    topologyConfig.put(com.twitter.heron.api.Config.TOPOLOGY_STMGRS, numContainers);
+
+    // Explicit set insufficient ram for container
+    long containerRam = -1L * Constants.GB;
+
+    topologyConfig.setContainerRamRequested(containerRam);
+
+    TopologyAPI.Topology topology =
+        getTopology(spoutParallelism, boltParallelism, topologyConfig);
+
+    PackingPlan packingPlan =
+        getResourceCompliantRRPackingPlan(topology);
+
+    Assert.assertNull(packingPlan);
```

This test should assert that a reasonable exception is thrown, instead of returning null